File: thrifts-dotnet/ThriftS.Service/LocalCache.cs
using System;
using System.Collections.Concurrent;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ThriftS.Service
{
    /// <summary>
    /// 本地进程内缓存
    /// </summary>
    internal static class LocalCache
    {
        /// <summary>
        /// 服务方法元数据
        /// </summary>
        public static readonly ConcurrentDictionary<string, Dictionary<string, ServiceMetaInfo>> ServiceDictionary =
            new ConcurrentDictionary<string, Dictionary<string, ServiceMetaInfo>>(StringComparer.OrdinalIgnoreCase);
    }
}

File: thrifts-java/thrifts-service/src/main/java/thrifts/service/LocalCache.java
package thrifts.service;

import java.util.TreeMap;

class LocalCache {
    public final static TreeMap<String,TreeMap<String,ServiceMetaInfo>> ServiceMap;

    static {
        ServiceMap = new TreeMap<String, TreeMap<String, ServiceMetaInfo>>(String.CASE_INSENSITIVE_ORDER);
    }
}
